### ISSUE TITLE
[frameworks: 15 llvm] fails with llvm/clang 14.0.0

### ISSUE THREAD

**Issue by lazka** [mesonbuild/meson#10249]
As pointed out here: https://github.com/mesonbuild/meson/pull/10248#issuecomment-1092454817

The error I see in the logs (`check_source_compiles: C: needs to be enabled before use.`) suggest it is https://github.com/llvm/llvm-project/issues/53950

And indeed

```diff
diff --git a/mesonbuild/dependencies/data/CMakeListsLLVM.txt b/mesonbuild/dependencies/data/CMakeListsLLVM.txt
index 9d3e41234..4084597d3 100644
--- a/mesonbuild/dependencies/data/CMakeListsLLVM.txt
+++ b/mesonbuild/dependencies/data/CMakeListsLLVM.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${C
 
 set(PACKAGE_FOUND FALSE)
 
+project(MesonTemp LANGUAGES C CXX)
+
 while(TRUE)
   find_package(LLVM REQUIRED CONFIG QUIET)
```

seems to make it work here.

I see that the generic CMakeLists file has some logic to inject the language, so I'm not sure if that should be extended or if the above patch is enough.

based on the file history -> pinging @mensinda

**Comment by mensinda** [comment]
Should be fixed with #10250.

**Comment by lazka** [comment]
thanks!
